packages/frontend/src/hooks/tableVisualization/useTableConfig.ts

```diff
@@ -107,6 +111,27 @@ const useTableConfig = (
         [columnProperties],
     );
 
+    const pivotTableData = useMemo<PivotData | undefined>(() => {
```

Comment:
I guess this pivotTable will always be triggered, even if the pivotTable is not requested. This could be very expensive on big tables,it would be nice if we can make this conditional, or lazy, or provide a callback to get on demand...

Reply:
yes ! you are right, I missed the `metricsAsRows` flag on the if. This should be ok then.